src/fontra/views/editor/panel-transformation.js

```diff
@@ -451,6 +530,77 @@ export default class TransformationPanel extends Panel {
     };
   }
 
+  async doPathOperations(pathOperationFunc, undoLabel) {
+    let { point: pointIndices } = parseSelection(this.sceneController.selection);
+    pointIndices = pointIndices || [];
+
+    const positionedGlyph =
+      this.sceneController.sceneModel.getSelectedPositionedGlyph();
+
+    const allContourIndices = [...range(positionedGlyph.glyph.path.numContours)];
+    const selectedContourIndices =
+      !pointIndices.length && undoLabel === "Remove overlaps"
+        ? allContourIndices
+        : [...getSelectionByContour(positionedGlyph.glyph.path, pointIndices).keys()];
```

Comment:
I want you to rewrite this, closer to how I originally suggested.

Idea:
```js
  const selectedContourIndicesMap = getSelectionByContour(positionedGlyph.glyph.path, pointIndices)
  const isContourSelected = pointIndices.length || !doUnion ? (i) => selectedContourIndicesMap.has(i) : (i) => true;
```

Then use`isContourSelected()` directly in the loop where you split the path in two.

Reply:
done.